### PR TITLE
Update cache tests for new interface

### PR DESCRIPTION
**Pull request by lleshchi**
Related #509
Expect some test failure until stratis-storage/stratisd#1969 is merged.

### CODE REVIEW

File: tests/whitebox/integration/pool/test_add.py [hunk 1]
```diff
@@ -51,19 +52,23 @@ def test_add(self):
...
-    Test adding devices to a non-existant pool.
+    Test adding cache devices before initializing cache.
     """
 
     _MENU = ["--propagate", "pool", "add-cache"]
     _POOLNAME = "deadpool"
 
+    def setUp(self):
+        super().setUp()
+        command_line = ["pool", "create", self._POOLNAME] + _DEVICE_STRATEGY()
```
Comment (jbaublitz) [inline tests/whitebox/integration/pool/test_add.py:63]: We need the `--propagate` flag here. That's why CI is failing.

Comment (jbaublitz) [reply tests/whitebox/integration/pool/test_add.py:63]: Disregard this. I need to do a little more digging.

**Comment by jbaublitz** [comment]
If your fix for the sim engine hasn't been merged yet, that would explain why this is failing.

**Comment by lleshchi** [comment]
Other possible tests:
-  Add devices to the cache of pool1, then add same devices to pool2
- Testing `add-cache` and `add-data` succeed when adding multiple devices
- Testing `add-cache` and `add-data` succeed when adding zero devices
- Initialize cache, add device to cache tier with `add-cache`, then attempt to add the device to the data tier with `add-data `(This might be redundant. There is a test to `init-cache` with a device, and then `add-data` with the same device.)
- Initialize pool, add device with `add-data`, then attempt to add the device to cache (This might also be redundant, since there is a test which initializes a pool with a device, then attempts to add that device to the cache)

**Comment by lleshchi** [comment]
Additionally, there are currently only 4 tests for` init-cache`. Many of the `add-cache` tests can be applied to `init-cache`, such as testing interactions between `init-cache` and `add-data`.

**Comment by mulkieran** [comment]
The tests look good so far. If you think there are more tests that are desirable to add go right ahead. Please add them individually and make sure you run each new test against your private copy of stratisd, to verify that all is as expected.

**Comment by mulkieran** [comment]
All the additional changes look fine to me. If you feel satisfied with the tests, and you've verified that the tests and coverage are passing against your private version of stratisd, then we should consider this ready.

Please add a link to the corresponding stratisd PR.

You can decide whether you want to squash some commits or not. I'm happy with them as they are. You might want to fix up a few commit messages, though, there's a typo: ```deviced```.

**Comment by mulkieran** [comment]
@lleshchi Please rebase.

**Pull request merged**